components/core/src/clp_s/ArchiveWriter.cpp

```
@@ -138,19 +142,104 @@
+     */
+    using schema_map_it = decltype(m_id_to_schema_writer)::iterator;
+    std::vector<schema_map_it> schemas;
+    std::vector<std::tuple<uint64_t, uint64_t>> stream_metadata;
+    std::vector<std::tuple<uint64_t, uint64_t, int32_t, uint64_t>> schema_metadata;
+
+    schema_metadata.reserve(m_id_to_schema_writer.size());
+    schemas.reserve(m_id_to_schema_writer.size());
+    for (auto it = m_id_to_schema_writer.begin(); it != m_id_to_schema_writer.end(); ++it) {
+        schemas.push_back(it);
+    }
+    auto comp = [](schema_map_it const& lhs, schema_map_it const& rhs) -> bool {
```

cppcheck

[error] Exception thrown in function declared not to throw exceptions.

(throwInNoexceptFunction)